docs/dev/opensearch-relevancy-search.md

```diff
@@ -0,0 +1,294 @@
+2. This design focuses on enabling users to do the search with SQL/PPL interface, but the analyzers and search algorithms are actually of the search engine job, so the features or enhancement of the search queries within the search engine are out of scope for this design.
+
+
+## 3 High Level Design
+
+### 3.1 Search functions to support
+
+Since the relevance based search is highly depending on the OpenSearch core engine, we are therefore defining the search functions following the existing search queries (see **appendix A1**) but in the language style of SQL full text search functions. Here comes the list of functions with basic functionalities (i.e. parameter options) to support as the relevance based search functions in the query engine:
+
+|Query Type	|Function Name In Query Engine	|Description	|
+|---	|---	|---	|
+|Match query	|match	|Returns documents that match a provided text, number, date or boolean value. The provided text is analyzed before matching.	|
+|Match phrase query	|match_phrase	|The `match_phrase` query analyzes the text and creates a `phrase` query out of the analyzed text.	|
+|Match phrase prefix query	|match_phrase_prefix	|Returns documents that contain the words of a provided text, in the **same order** as provided. The last term of the provided text is treated as a prefix, matching any words that begin with that term.	|
+|Match boolean prefix query	|match_bool_prefix	|analyzes its input and constructs a `bool` query from the terms. Each term except the last is used in a `term` query. The last term is used in a `prefix` query.	|
+|Multi match query	- best fields	|multi_match	|Finds documents which match any field, but uses the `_score` from the best field.	|
+|Multi match query	- most fields	|multi_match	|Finds documents which match any field and combines the `_score` from each field.	|
+|Multi match query	- cross fields	|multi_match	|Treats fields with the same `analyzer` as though they were one big field. Looks for each word in **any** field.	|
+|Multi match query	- phrase	|multi_match	|Runs a `match_phrase` query on each field and uses the `_score` from the best field.	|
+|Multi match query	- phrase prefix	|multi_match	|Runs a `match_phrase_prefix` query on each field and uses the `_score` from the best field.	|
+|Multi match query	- bool prefix	|multi_match	|Creates a `match_bool_prefix` query on each field and combines the `_score` from each field. 	|
+|Combined fields 	|combined_fields	|analyzes the query string into individual terms, then looks for each term in any of the fields	|
+|Common terms query	|common	|The `common` terms query is a modern alternative to stopwords which improves the precision and recall of search results (by taking stopwords into account), without sacrificing performance.	|
```

Comment:
`combined_fields` and `common` doesn't exist in SQL plugin. Furthermore, `combined_fields` doesn't exist in OpenSearch: https://github.com/opensearch-project/OpenSearch/issues/3996.
`common_terms` is deprecated: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/query/CommonTermsQueryBuilder.java#L64
```suggestion
```